### DISCUSSION TITLE
merge items with durabilities

### DISCUSSION THREAD

**Discussion by scarf005** [cataclysmbn/Cataclysm-BN#3057] [General]
[discussion in discord](https://discord.com/channels/830879262763909202/830916451517857894/1140528550240337920)

[![](https://steamuserimages-a.akamaihd.net/ugc/785233007824414020/8E0576151EDDB7EB050FD939727F349B74888777/?imw=637&imh=358&ima=fit&impolicy=Letterbox&imcolor=%23000000&letterbox=true)][steam]
[optimization mod with the same concept in don't starve together][steam]

[steam]: https://steamcommunity.com/sharedfiles/filedetails/?id=1803285852

- if there are `N` items of same id in same tile, convert them into 1 item of `N` 'count's.
- optimizes moving 100,000 felt patches around
- average durability? not sure, maybe patches should lose charges instead of losing durabilities when damaged

should be worked on top of #2250 . maybe this idea was already discussed before?

**Comment by ghost** [comment]
The way I see this for what I will refer to as the "trouble cases", or basically things with data.

Melee weapons and tools: You have 20 longswords with 11 durability, 3 with 1/ durability, and so on. Merge the ones with the same durability and mods. Players can name them if they want em separated.

Guns: Can't be done IMO unless they have 0 fouling and faults, empty or full magazine. You have 30 M4A1's with different foulings durabilities magazines attachments etc. They could only be merged if they shared all the same properties and thus generally would only be practical if it was pristine.

Food: Anything nonperishable with the same ingredients. I think for food it would be better if there was a function to merge nonperishable foods into a singular "Food" item, which was a uniform calorie value and set weight and volume. Players could convert 40 different types of nonperishable foodstuffs into one type and save on memory at the cost of realism/immersion and this would thus be an optional function. Calorie values won't easily line up to the set value every time so there'd need to be a way to leave a scrap item of the remaining calories.

**Reply by scarf005** [reply]
> Food: Anything nonperishable with the same ingredients. I think for food it would be better if there was a function to merge nonperishable foods into a singular "Food" item, which was a uniform calorie value and set weight and volume.

could probably add a ['tinning kit'](https://nethackwiki.com/wiki/Tinning_kit) mod that allows player to do this.
